src/js/core.js

```diff
@@ -42,8 +42,13 @@ function MediumEditor(elements, options) {
             // If Shift is down, outdent, otherwise indent
             if (event.shiftKey) {
                 this.options.ownerDocument.execCommand('outdent', false, null);
+                Util.cleanListDOM(this.options.ownerDocument, Selection.getSelectionStart(this.options.ownerDocument));
             } else {
-                this.options.ownerDocument.execCommand('indent', false, null);
+                // If it's the first element in the list, ignore this command
+                if (node.previousSibling !== null) {
+                    this.options.ownerDocument.execCommand('indent', false, null);
```

Comment:
I tried using `exportSelection` and `importSelection` here, calling export before the command is triggered and import after the call to `cleanListDOM`, but unfortunately that did not work as hoped. In fact, the browser behavior jumped back to the original uncorrected behavior sending the cursor to the end of the visual line.